src/header.ts

```diff
@@ -0,0 +1,45 @@
+import { BufferReader, Hash } from "./utils";
+
+export default class Header {
+  version: Buffer;
+  prevHash: Buffer;
+  merkleRoot: Buffer;
+  time: number;
+  bits: Buffer;
+  nonce: number;
+  buffer: Buffer;
+  hash?: Buffer;
+
+  private constructor(br: BufferReader) {
```

Comment:
Private constructors should work well here since we want to enforce the static initializers.